modules/apps/site/site-memberships-web/src/main/java/com/liferay/site/memberships/web/internal/servlet/taglib/util/UserActionDropdownItemsProvider.java

```diff
@@ -52,6 +52,28 @@ public List<DropdownItem> getActionDropdownItems() throws Exception {
 				_themeDisplay.getSiteGroupIdOrLiveGroupId(),
 				ActionKeys.ASSIGN_USER_ROLES),
 			_getAssignRolesActionUnsafeConsumer()
+		).add(
+			dropdownItem -> {
+				dropdownItem.putData("action", "unassignRoles");
+				dropdownItem.putData(
+					"unassignUserRoleURL",
+					PortletURLBuilder.createActionURL(
+						_renderResponse
+					).setMVCPath(
+						"/users_roles.jsp"
+					).setParameter(
+						"assignRoles", Boolean.FALSE
+					).setParameter(
+						"groupId", _themeDisplay.getSiteGroupIdOrLiveGroupId()
+					).setParameter(
+						"p_u_i_d", _user.getUserId()
```

Comment:
Is it necessary to set the `p_u_i_d`? I think this is only used for `_portal.getSelectedUser` but we're setting the `userId` explicitly.

Reply:
@jonathanmccann I assume this is needed, since the `userId` that is set above it is needed just to the `unassignRoles` method in `actions.js`, while, as you said, the `p_u_i_d` is needed in `_portal.getSelectedUser`.